### DISCUSSION TITLE
Query works with RLS disabled but does not work with RLS enabled, even though I am using the service key.

### DISCUSSION THREAD

**Discussion by adrfinance** [supabase/supabase#30299] [Questions]
Hello,
I am using Buildship for my back end, to make a call to Supabase which is my DB.

The code works when RLS is disabled.

But when I enable RLS, the results is an empty array ([]) even though I am using the service key which is supposed to bypass RLS?! 

What am I doing wrong?

I am using the following code:

`
import fetch from 'node-fetch';

export default async function getSupabaseTableRow(
  { supabaseUrl, table, filter }: NodeInputs,  // Node input parameters
  { logging, env }: NodeScriptOptions         // Node options for logging and environment variables
): NodeOutput {
  
    // Access the Supabase service role key securely from environment variables
    const supabaseKey = env.get('SUPABASE_SERVICE_ROLE_KEY', 'project');

    if (!supabaseKey) {
        logging.log("Service role key is missing in environment variables.");
        throw new Error("Service role key is not configured.");
    }

    // Construct the API endpoint URL
    const url = `${supabaseUrl}/rest/v1/${table}?${encodeURIComponent(filter)}`;

    logging.log(`Fetching from URL: ${url}`);  // Log the URL being fetched
    logging.log(`Using filter: ${filter}`);     // Log the filter being used

    const headers = {
        'apikey': supabaseKey,  // Use the service role key from env
    };

    const requestOptions = {
        method: 'GET',
        headers
    };

    try {
        const response = await fetch(url, requestOptions);

        if (!response.ok) {
            logging.log(`Error fetching data from table ${table}: ${response.statusText}`);
            const errorBody = await response.text(); // Log response body if not ok
            logging.log(`Response body: ${errorBody}`);
            throw new Error(`Error fetching data: ${response.statusText}`);
        }
        
        const data = await response.json();
        logging.log(`Data successfully fetched from table ${table}: ${JSON.stringify(data)}`);
        
        return data;
    } catch (error) {
        logging.log(`Fetch failed: ${error.message}`);
        throw error;
    }
}
`

**Comment by GaryAustin1** [comment]
You are not service_role if RLS blocks you.  You likely are setting a user session in the authorization header either on your own or by using SSR or auth signin/up.   A user session will override the default use of the apikey in the authorization header.

**Reply by GaryAustin1** [reply]
https://github.com/orgs/supabase/discussions/30146
